src/components/UpupUploader/FileBrowser/BrowserOD.tsx

```diff
@@ -14,59 +16,269 @@ type Props = {
+                    ...file,
+                    '@microsoft.graph.downloadUrl':
+                        fileInfo['@microsoft.graph.downloadUrl'],
+                    thumbnails: fileInfo.thumbnails?.[0] || null,
+                    file: fileInfo.file,
+                }
+
+                console.log('Updated file object:', updatedFile)
+
+                setSelectedFiles(prevFiles => {
+                    const newFiles = prevFiles.includes(file)
+                        ? prevFiles.filter(f => f.id !== file.id)
+                        : [...prevFiles, updatedFile]
+                    console.log('Updated selected files:', newFiles)
+                    return newFiles
+                })
+            } catch (error) {
+                console.error(
+                    `Error fetching file information for ${file.name}:`,
+                    error,
+                )
+            }
         }
+    }, 500)
+
+    const getDownloadUrl = async (file: OneDriveFile, graphClient: Client) => {
+        // First try to get direct download URL
+        const fileInfo = await graphClient
+            .api(`/me/drive/items/${file.id}`)
+            .select('@microsoft.graph.downloadUrl')
+            .get()
+
+        if (fileInfo['@microsoft.graph.downloadUrl']) {
+            return fileInfo['@microsoft.graph.downloadUrl']
+        }
+
+        // Fallback to creating a sharing link
+        console.log('No direct download URL, creating sharing link...')
+        const permission = await graphClient
+            .api(`/me/drive/items/${file.id}/createLink`)
+            .post({
+                type: 'view',
+                scope: 'anonymous',
+            })
+
+        const shareId = permission.link.webUrl.split('/s!/')[1]
+        const sharedItem = await graphClient
+            .api(`/shares/u!${shareId}`)
+            .expand('driveItem')
+            .get()
+
+        return sharedItem.driveItem['@microsoft.graph.downloadUrl']
+    }
+
+    const downloadFile = async (
+        url: string,
+        fileName: string,
+        mimeType = 'application/octet-stream',
+    ) => {
+        const response = await fetch(url, {
+            method: 'GET',
+        })
+
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`)
+        }
+
+        const blob = await response.blob()
+        return new File([blob], fileName, {
+            type: mimeType || 'application/octet-stream',
+        })
     }
 
     const downloadFiles = async (files: OneDriveFile[]) => {
-        const promises = files.map(async file => {
-            const data = await downloadFile(
-                file['@microsoft.graph.downloadUrl']!,
-            )
-            const downloadedFile = new File([data], file.name, {
-                type: file.file?.mimeType,
-            }) as unknown as OneDriveFile
+        const promises = files.map(async (file, index) => {
+            console.log('Processing file:', file)
+
```

Comment:
Make sure to clean all the code from console.logs

Reply:
Thanks @MedAmine1212 , done